src/main/java/com/amazon/opendistroforelasticsearch/sql/domain/Select.java

```diff
@@ -83,40 +86,41 @@ public void setHaving(Having having) {
         this.having = having;
     }
 
-	public List<Order> getOrderBys() {
-		return orderBys;
-	}
+    public List<Order> getOrderBys() {
+        return orderBys;
+    }
 
-	public int getOffset() {
-		return offset;
-	}
+    public int getOffset() {
+        return offset;
+    }
 
-	public int getRowCount() {
-		return rowCount;
-	}
+    public int getRowCount() {
+        return rowCount;
+    }
 
-	public void addOrderBy(String nestedPath, String name, String type, boolean isScriptField) {
-		if ("_score".equals(name)) {
-			isQuery = true;
-		}
-		this.orderBys.add(new Order(nestedPath, name, type, isScriptField));
-	}
+    public void addOrderBy(String nestedPath, String name, String type, boolean isScriptField) {
+        if ("_score".equals(name)) {
+            isQuery = true;
+        }
+        this.orderBys.add(new Order(nestedPath, name, type, isScriptField));
+    }
 
 
-	public void addField(Field field) {
-		if (field == null ) {
-			return;
-		}
-        if(field.getName().equals("*")){
+    public void addField(Field field) {
+        if (field == null ) {
+            return;
+        }
+        if (field == STAR && !isAgg) { // Ignore GROUP BY since columns present in result are decided by column list in GROUP BY
             this.selectAll = true;
+            return;
         }
 
-		if(field instanceof  MethodField && aggsFunctions.contains(field.getName().toUpperCase())) {
-			isAgg = true;
-		}
+        if(field instanceof  MethodField && aggsFunctions.contains(field.getName().toUpperCase())) {
```

Comment:
formatting: missing space between if and (

Reply:
done!